text/0497-glue-l2-construct.md

```diff
@@ -0,0 +1,410 @@
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-python-script'),
+   extraJarsS3Url: [glue.Code.fromBucket('bucket-name', 'path-to-extra-scala-jar')],
+   description: 'an example pySpark ETL job',
+   numberOfWorkers: 20
+});
+```
+
+2. **Streaming Jobs**
+
+A Streaming job is similar to an ETL job, except that it performs ETL on data
+streams using the Apache Spark Structured Streaming framework. Some Spark
+job features are not available to streaming ETL jobs. These jobs will default
+to use Python 3.9.
+
+Similar to ETL streaming job supports Scala and Python languages. Similar to ETL,
+it supports G1 and G2 worker type and 2.0, 3.0 and 4.0 version. We’ll default
+to G2 worker and 4.0 version for streaming jobs which developers can override.
+We will enable `—enable-metrics, —enable-spark-ui, —enable-continuous-cloudwatch-log`.
+
+```
+new glue.PythonSparkStreamingJob(this, 'PythonSparkStreamingJob', {
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-python-script'),
+});
+
+
+new glue.ScalaSparkStreamingJob(this, 'ScalaSparkStreamingJob', {
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-scala-jar'),
+   className: 'com.example.HelloWorld',
+});
+
+```
+
+Optionally, developers can override the glueVersion and add extraJars and a
+description:
+
+```
+new glue.PythonSparkStreamingJob(this, 'PythonSparkStreamingJob', {
+   glueVersion: glue.GlueVersion.V3_0,
+   pythonVersion: glue.PythonVersion.3_6,
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-python-script'),
+   description: 'an example Python Streaming job',
+   numberOfWorkers: 20,
+});
+
+
+new glue.ScalaSparkStreamingJob(this, 'ScalaSparkStreamingJob', {
+   glueVersion: glue.GlueVersion.V3_0,
+   pythonVersion: glue.PythonVersion.3_6,
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-scala-jar'),
+   className: 'com.example.HelloWorld',
+   description: 'an example Python Streaming job',
+   numberOfWorkers: 20,
+});
+```
+
+3. **Flex Jobs**
+
+The flexible execution class is appropriate for non-urgent jobs such as
+pre-production jobs, testing, and one-time data loads. Flexible job runs
+are supported for jobs using AWS Glue version 3.0 or later and `G.1X` or
+`G.2X` worker types but will default to the latest version of Glue
+(currently Glue 3.0.) Similar to ETL, we’ll enable these features:
+`—enable-metrics, —enable-spark-ui, —enable-continuous-cloudwatch-log`
+
+```
+glue.ScalaSparkFlexEtlJob(this, 'ScalaSparkFlexEtlJob', {
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-scala-jar'),
+   className: 'com.example.HelloWorld',
+});
+
+glue.pySparkFlexEtlJob(this, 'pySparkFlexEtlJob', {
+   script: glue.Code.fromBucket('bucket-name', 's3prefix/path-to-python-script'),
+});
+```
+
+Optionally, developers can override the glue version, python version,
+provide extra jars, and a description
+
+```
+glue.ScalaSparkFlexEtlJob(this, 'ScalaSparkFlexEtlJob', {
```

Comment:
Looks like pythonVersion is missing from this one. Are these the complete set of props proposed?

Reply:
Scala jobs doesn't require python version which is why didn't specified

Reply:
Got it. Thanks for the clarification!